File: SideQuest/SideQuest-FrontEnd/src/types/Projeto.ts
// Projeto
export type StatusProjeto = 'ATIVO' | 'EM_ANDAMENTO' | 'ATRASADO' | 'CONCLUIDO' | 'CONCLUIDO_COM_ATRASO';

export type Projeto = {
  id: string;
  nome: string;
  status: StatusProjeto;
  usuarioIds: string[];
  prazo: string;
  descricao?: string;
}

File: SideQuest/SideQuest-BackEnd/projetos-service/src/main/java/com/syntax/projetos_service/modelo/entidade/Projeto.java
package com.syntax.projetos_service.modelo.entidade;

import java.util.Date;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

/**
 * ENTIDADE PROJETO
 * 
 * Representa um projeto no sistema.
 * Armazenada na collection "projetos" do MongoDB.
 */
@Data
@Document(collection = "projetos")
public class Projeto {
    @Id
    private String id;
    private String status;
    private String nome;
    private String descricao;
    private Date prazoFinal;
    private List<String> usuarioIds;
}
